### ISSUE TITLE
ERROR Operation timed out when attempting keybase device add

### ISSUE THREAD

**Issue by nealevans** [keybase/client#5852]
Hello, I've been trying to install the new keybase client on my Mac. When the client loads, I select 'Log In' under Already on Keybase? Then I enter my username, select the device I want to connect with (my laptop was in the list), and the client asked that I enter the following in a shell:

keybase device add

When I do this, I get the following response after a few seconds:
ERROR Operation timed out

I tried completely uninstalling and reinstalling.

Log ID is e1efac052d66d233e9465f1c.

Thank you!

**Comment by maxtaco** [comment]
Did you run `keybase device add` on your *other device* or was it on the same device?

**Comment by nealevans** [comment]
Same device. I never installed keybase on any other device.

**Comment by maxtaco** [comment]
yeah, that's the problem, but you're not the first person to have made this mistake (the instructions say to run that command on your **other** device)

Cc: @malgorithms  on this

**Comment by nealevans** [comment]
Wait, what other device? I've only ever had keybase on this laptop.

**Comment by nealevans** [comment]
I see now it says Run keybase device add on your other device and then type this code in here.

I missed that part about the other device; nevertheless, I'm not sure what that is referring to. Does this mean run the comment on a completely different instance, e.g. linux running in a VM?

**Comment by patrickxb** [comment]
@nealevans you have one physical device currently registered, a computer you named 'wopr':

https://keybase.io/nealevans/devices

Are you trying to run keybase on that computer, or on a different computer?

**Comment by malgorithms** [comment]
I'm a bit confused by this conversation.

@nealevans  - typically you're only ever asked to do that command when installing keybase on a 2nd device.

Example: 

1. I install keybase on "imac" and login with my username and password.
2. Later, I decide I want to install keybase on a different machine, "macbook". When I enter my username, Keybase knows that I've already provisioned a device called "imac"; at this point, the only way to add a computer is to have the old one sign it in.
3. So, device 2 (macbook) asks me to run something on device 1 (imac).

This is the first time I've seen this confusion.

Perhaps, you did something like this:

1. installed on your mac (`wopr` as patrick said)
2. uninstalled keybase on your mac, or factory reset it, losing your keybase device key?
3. then installed again on the same mac? (It doesn't know it's the same mac, so it's asking you to go to it and type something on `wopr` to prove you're you)

Thanks for your help on this...would like to get to the bottom of it!

**Comment by nealevans** [comment]
Ah, I think I see what is going on, perhaps. This machine _is_ wopr:

➜  ~ uname -n
wopr.local

One thing that has happened since I installed keypads back on Sep 6 is that did an 'in situ' downgrade from a beta of macOS to the current 10.12.n. It wasn't a wipe and reinstall, but I installed 'over' the beta macOS (thereby keeping my apps and docs). That must have confused keybase somehow.

What is the typical output of 'keybase paperkey'? I'm trying to figure out where I might have saved off that key.

If I can't find that paperkey, do I have any other recovery options?

**Comment by malgorithms** [comment]
if you've lost your paper key (perhaps it's in a drawer/wallet/purse? it's 13 words long, I believe) *and* the device key somehow got lost in this OS version downgrade, then there's nothing we can do on our end...since your 2 keys are never on our servers.  

In that case the only thing to do would be an account reset (accessible somewhere in that login menu), although to be clear, you would lose any data you have in KBFS or chat, and you would need to repost your twitter/whatever proofs.

So if you have that paper key written down somewhere, that's the best option.

I don't know anything about a macOS version downgrade or how that would affect Keybase's key store. If you do have data you're afraid to lose, then perhaps @gabriel might have an idea on whether your local key is recoverable. But that might be a pain, so if you don't have data in there, and you can't find your paper key, I would just advise the account reset.

**Comment by patrickxb** [comment]
@nealevans what do you see in ~/Library/Application Support/Keybase?  Is there a config.json file in there and a secretkeys.nealevans.mpack file?

**Comment by nealevans** [comment]
@patrickxb, ~/Library/Application Support/Keybase doesn't exist.

@malgorithms, thanks for feedback. I haven't used kbfs or chat, so I don't have any data I have to worry about losing. Sounds like account reset and re-proofing is best best. I remember creating that paper key, but I didn't save it out (didn't follow my SOPs).

Good learning experience. Thanks guys!

**Comment by nealevans** [comment]
Reset and restart working.

One little bug perhaps: even after reset and restart, I couldn't re-add device 'wopr'. I said it already existed. When I went in web UI to try to delete it, no device were listed.

To workaround, I just added 'wopr-2'.
